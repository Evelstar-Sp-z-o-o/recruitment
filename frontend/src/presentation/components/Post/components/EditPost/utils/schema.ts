import * as yup from 'yup';

const schema = yup.object().shape({
  post: yup
    .string()
    .min(10, 'The minimum number of characters is 10')
    .max(350, 'The maximum number of characters is 350'),
});

export default schema;
