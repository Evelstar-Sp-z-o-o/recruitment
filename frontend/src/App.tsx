import React from 'react';
import { BrowserRouter as Router, Routes, Route, Link } from 'react-router-dom';

import { AppBar, Toolbar, Typography, Container } from '@mui/material';

import EditPost from './components/EditPost';
import PostForm from './components/PostForm';
import PostList from './components/PostList';

const App: React.FC = () => {
  return (
    <Router>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6">
            <Link to="/" style={{ color: '#fff', textDecoration: 'none' }}>
              Home
            </Link>
          </Typography>
          <Typography variant="h6" style={{ marginLeft: 'auto' }}>
            <Link to="/create" style={{ color: '#fff', textDecoration: 'none' }}>
              Create Post
            </Link>
          </Typography>
        </Toolbar>
      </AppBar>
      <Container>
        <Routes>
          <Route path="/" element={<PostList />} />
          <Route path="/create" element={<PostForm />} />
          <Route path="/edit/:id" element={<EditPost />} />
        </Routes>
      </Container>
    </Router>
  );
};

export default App;
