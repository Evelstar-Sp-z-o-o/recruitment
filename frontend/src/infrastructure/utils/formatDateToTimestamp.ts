const formatDateToTimestamp = () => {
  const today = new Date();

  const unixTimestamp = Math.floor(today.getTime() / 1000);

  return unixTimestamp;
};

export default formatDateToTimestamp;
