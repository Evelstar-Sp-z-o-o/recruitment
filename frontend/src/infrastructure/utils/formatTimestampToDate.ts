import formatDateComponents from './formatDateComponents';

const formatTimestampToDate = (timestamp: number) => {
  const date = new Date(timestamp * 1000);

  const { day, month, year, hours, minutes } = formatDateComponents(date);

  const formattedDate = `${day}.${month}.${year} ${hours}:${minutes}`;

  return formattedDate;
};

export default formatTimestampToDate;
