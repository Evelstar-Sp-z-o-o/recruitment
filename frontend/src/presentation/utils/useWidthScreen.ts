import { useMediaQuery, useTheme } from '@mui/material';

const useWidthScreen = () => {
  const theme = useTheme();

  const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'));

  return { isSmallScreen };
};

export default useWidthScreen;
