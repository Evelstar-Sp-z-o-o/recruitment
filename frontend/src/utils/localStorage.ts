import { User } from '../types';

const USER_KEY = 'loggedUser';

export const saveUserToLocalStorage = (user: User) => {
  localStorage.setItem(USER_KEY, JSON.stringify(user));
};

export const getUserFromLocalStorage = () => {
  const user = localStorage.getItem(USER_KEY);
  return user ? JSON.parse(user) : null;
};

export const removeUserFromLocalStorage = () => {
  localStorage.removeItem(USER_KEY);
};
