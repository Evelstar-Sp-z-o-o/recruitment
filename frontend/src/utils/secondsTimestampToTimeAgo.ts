export const secondsTimestampToTimeAgo = (timestamp: number): string => {
  const now = new Date();
  const createdDate = new Date(timestamp * 1000); // convert seconds to milliseconds
  const diff = now.getTime() - createdDate.getTime(); // difference in milliseconds

  if (diff < 1) {
    return 'now';
  }

  const seconds = Math.floor(diff / 1000);
  const minutes = Math.floor(seconds / 60);
  const hours = Math.floor(minutes / 60);
  const days = Math.floor(hours / 24);
  const weeks = Math.floor(days / 7);

  if (weeks > 0) {
    return `${weeks} week${weeks > 1 ? 's' : ''} ago`;
  } else if (days > 0) {
    return `${days} day${days > 1 ? 's' : ''} ago`;
  } else if (hours > 0) {
    return `${hours} hour${hours > 1 ? 's' : ''} ago`;
  } else if (minutes > 0) {
    return `${minutes} minute${minutes > 1 ? 's' : ''} ago`;
  } else {
    return `${seconds} second${seconds > 1 ? 's' : ''} ago`;
  }
};
