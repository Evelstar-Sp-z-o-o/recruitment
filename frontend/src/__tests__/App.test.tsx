import { Provider } from 'react-redux';

import App from '@/src/App';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import { createTestStore } from './utils';

describe('App Component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={createTestStore()}>
        <App />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // Testowanie renderowania linków
  it('should render navigation links correctly', () => {
    const { getByText } = render(
      <Provider store={createTestStore()}>
        <App />
      </Provider>,
    );

    // Sprawdzenie, czy linki są obecne
    expect(getByText('Home')).toBeInTheDocument();
    expect(getByText('Create Post')).toBeInTheDocument();
  });
});
