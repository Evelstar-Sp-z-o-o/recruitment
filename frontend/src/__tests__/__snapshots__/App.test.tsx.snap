// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        <a
          href="/"
          style="color: rgb(255, 255, 255); text-decoration: none;"
        >
          Home
        </a>
      </h6>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        style="margin-left: auto;"
      >
        <a
          href="/create"
          style="color: rgb(255, 255, 255); text-decoration: none;"
        >
          Create Post
        </a>
      </h6>
    </div>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div />
  </div>
</DocumentFragment>
`;
